### PR TITLE
Update .clang-format formatting of function declaration parameters

### PR DESCRIPTION
**Pull request by tristan0x**
@pramodk @ferdonline and I agree that function declaration should be specified in one line if declaration fits. Otherwise, there should have one parameter per line.

### CODE REVIEW

**Comment by tristan0x** [comment]
@pramodk @ferdonline Only last commit is relevant to this PR. 2 first are for documentation generation.

**Comment by tristan0x** [comment]
Please review @pramodk and @ferdonline

File: cpp/formatting/snippets/one-parameter-per-line-if-they-do-no-fit.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// One parameter per line if they don't all fit
+// clang-format: AllowAllParametersOfDeclarationOnNextLine: false
+// clang-format: BinPackParameters: False
+
+void andAVeryLongFunctionName(the_long_parameter1,
+                              the_long_parameter2,
+                              the_long_parameter3,
+                              the_long_parameter4);
+
+void andALongFunctionName(the_long_parameter1, the_long_parameter2);
```
Comment (pramodk) [inline cpp/formatting/snippets/one-parameter-per-line-if-they-do-no-fit.cpp]: I prefer this!

Comment (ferdonline) [reply cpp/formatting/snippets/one-parameter-per-line-if-they-do-no-fit.cpp]: `the_long_parameter1` is not a long parameter.
From a real world thing :)
```
HighFive::DataSet dump(HighFive::File& file,
                       const std::string& path,
                       const std::vector<T>& data,
                       HighFive::Mode mode = HighFive::Mode::Create);
```
Anyway, I totally agree with the change (I even had suggested that in the gerrit review)

Comment (tristan0x) [reply cpp/formatting/snippets/one-parameter-per-line-if-they-do-no-fit.cpp]: I changed with your HighFive example.

**Comment by tristan0x** [comment]
your review has been discarded. Can you please accept again?

File: cpp/formatting/README.md [hunk 2]
```diff
@@ -30,8 +30,13 @@ A C++ snippet has the following structure:
...
+// clang-format: KEY: VALUE
+// clang-format: KEY: VALUE
+
 
 template <typename T>
 your cpp(code);
 ```
-If a C++ snippet highlights a particular clang-format configuration key, then it should be named after it.
+
+If a C++ snippet highlights a particular clang-format configuration key, then it should be named after it. Additional Clang-Format key/value
```
Comment (matz-e) [inline cpp/formatting/README.md]: Really long line. Text should break at 80 chars!

Comment (ferdonline) [reply cpp/formatting/README.md]: It's breaking only code?

Comment (ferdonline) [reply cpp/formatting/README.md]: Agree. we should break just like code. @matz-e would you agree using 100 chars for everything?

Comment (tristan0x) [reply cpp/formatting/README.md]: > Really long line. Text should break at 80 chars!
You are right.
This README.md is not enforced by C++ formatting rules.

Comment (tristan0x) [reply cpp/formatting/README.md]: > Agree. we should break just like code. @matz-e would you agree using 100 chars for everything?

I would prefer to have 80 columns, but 100 is more pragmatic.

Comment (matz-e) [reply cpp/formatting/README.md]: > > Agree. we should break just like code. @matz-e would you agree using 100 chars for everything?
> 
> I would prefer to have 80 columns, but 100 is more pragmatic.

Same. This is Markdown, so to me that's an editor setting. In code documentation should be wrapped shorted than code, potentially for me. If that's not an option, I can live with that.

**Pull request merged**